### PR TITLE
[node] v16 types

### PR DESCRIPTION
**Pull request by victorperin**
### 🚧 🚧 🚧 Work In Progress 🚧 🚧 🚧 

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/node/timers/promises.d.ts [hunk 1]
```diff
@@ -14,4 +14,9 @@ declare module 'timers/promises' {
      * Returns a promise that resolves in the next tick.
      */
     function setImmediate<T>(value: T, options?: TimerOptions): Promise<T>;
+
+    /**
+     * Returns an async iterator that generates values in an interval of delay ms.
+     */
+    function setInterval<T>(delay: number, value?: T, options?: TimerOptions): Promise<T>;
```
Comment (Mesteery) [inline types/node/timers/promises.d.ts:21]:
```suggestion
    function setInterval<T>(delay: number, value?: T, options?: TimerOptions): AsyncIteratorIterable<T>;
```

**Comment by Mesteery** [comment]
sorry, the true type is `AsyncIterableIterator`. (and don't forget to update tests)

**Comment by gperdomor** [comment]
@victorperin thanks for your work, i hope this can be merged soon :D

**Comment by ExE-Boss** [comment]
Since **Node v16** supports the `node:` prefix in both **ESM** and **CommonJS**, <https://github.com/DefinitelyTyped/DefinitelyTyped/pull/51107> should be re‑introduced (by reverting <https://github.com/DefinitelyTyped/DefinitelyTyped/pull/52595> in **v16**)

**Comment by SimonSchick** [comment]
I have a series of PRs out which should probably be merged first, otherwise this could get pretty annoying updating v15 and 16.

**Comment by karlhorky** [comment]
@SimonSchick Which PRs are in this series? I guess these will all be done in a week or two?

**Comment by SimonSchick** [comment]
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/52732 https://github.com/DefinitelyTyped/DefinitelyTyped/pull/52775 and https://github.com/DefinitelyTyped/DefinitelyTyped/pull/52776 and one more PR that is not out yet.

I hope to have 15.6 in end of week, unfortunately 15.3 took a significant amount of time (16 days) to be approved so I'd assume the entire will take at least until the end of the month.

**Comment by karlhorky** [comment]
@SimonSchick looks like the `node:` scheme / prefix was backported to Node.js 14.13.1 / 12.20.0 🎉

Maybe you could also do some new PRs for the v14 and v12 lines re-introducing the change by @ExE-Boss ?

Reference: https://nodejs.org/api/esm.html -> link "node: Imports" (links with anchors don't scroll properly right now)

Screen Shot 2021-05-30 at 11 04 39

**Comment by SimonSchick** [comment]
My final v15 PR is probably going to land this week https://github.com/DefinitelyTyped/DefinitelyTyped/pull/53566
We can more actively work on this now, I think.

**Comment by SimonSchick** [comment]
Please make sure to redo the file copying.

**Comment by SimonSchick** [comment]
@victorperin, do you want to drive this PR or do you mind me to take over?

I realized I created a v16 branch a while ago (before I saw this PR), I think I could re-use some parts of this PR as well.

**Comment by victorperin** [comment]
Oh, no problem, I'm currently under a ton of work latelly, i think you could end this faster than me. Thanks!

**Comment by karlhorky** [comment]
`@types/node@16.0.0` has been published now - the `node:` schema / prefix now works again:

- https://www.npmjs.com/package/@types/node
- https://github.com/DefinitelyTyped/DefinitelyTyped/pull/53669

**Pull request closed without merging**